app/v2/connectors/V2BaseConnector.scala

```diff
@@ -145,6 +140,31 @@ trait V2BaseConnector extends HttpErrorFunctions {
         case None         => requestBuilder
       }
 
+    def withMovementId(movementId: Option[MovementId]): RequestBuilder =
+      if (movementId.isDefined)
+        requestBuilder.setHeader("X-Audit-Meta-Movement-Id" -> movementId.get.value)
+      else requestBuilder
+
+    def withEoriNumber(eoriNumber: Option[EORINumber]): RequestBuilder =
+      if (eoriNumber.isDefined)
+        requestBuilder.setHeader("X-Audit-Meta-EORI" -> eoriNumber.get.value)
+      else requestBuilder
+
+    def withMovementType(movementType: Option[MovementType]): RequestBuilder =
+      if (movementType.isDefined)
+        requestBuilder.setHeader("X-Audit-Meta-Movement-Type" -> movementType.get.movementType)
+      else requestBuilder
+
+    def withMessageType(messageType: Option[MessageType]): RequestBuilder =
+      if (messageType.isDefined)
+        requestBuilder.setHeader("X-Audit-Meta-Message-Type" -> messageType.get.code)
+      else requestBuilder
+
+    def withMessageId(messageId: Option[MessageId]): RequestBuilder =
+      if (messageId.isDefined)
+        requestBuilder.setHeader("X-Audit-Meta-Message-Id" -> messageId.get.value)
+      else requestBuilder
```

Comment:
Now you've done this (and I approve!) we can test these in isolation too -- let's make a set of unit tests for these alone.